imap/imapd.c

```diff
@@ -5004,10 +5009,13 @@ static void cmd_close(char *tag, char *cmd)
     }
 
     /* local mailbox */
-    if ((cmd[0] == 'C' || config_getswitch(IMAPOPT_AUTOEXPUNGE)) && index_hasrights(imapd_index, ACL_EXPUNGE)) {
+    if (cmd[0] == 'C') {
```

Comment:
This looks like a mistake - now we'll call `index_expunge(imapd_index, NULL, 1)` on `close` without checking if they have EXPUNGE rights, when before we'd check 

If I open two ctel sessions, and do  `x SELECT INBOX` and then `x STORE 1 +Flags \Deleted` in one, and then in the other I do `x EXAMINE INBOX` and `x CLOSE`, imapd crashes with:
```
2024-12-10T10:02:19.351166-05:00 box-fastmail-01 (err) slotf1n01/imapadmin[3149412]:  Internal error: assertion failed (aborting): imap/mailbox.c: 4516: mailbox_index_islocked(mailbox, 1)

```